src/tokens/tokens.service.ts

```diff
@@ -129,101 +155,144 @@ export class TokensService {
-          amount: dto.amount,
-          data: encodeHex(dto.data ?? ''),
-          to: dto.to,
-        },
+          headers: {
+            type: sendTransactionHeader,
+          },
+          from: dto.operator,
+          to: validPoolId.address,
+          method: methodAbi,
+          params: [dto.to, dto.amount, encodeHex(dto.data ?? '')],
+        } as EthConnectMsgRequest,
         this.postOptions(dto.operator, dto.requestId),
       ),
     );
+
     return { id: response.data.id };
   }
 
   async transfer(dto: TokenTransfer): Promise<AsyncResponse> {
+    const validPoolId: ITokenPool = this.validatePoolId(new URLSearchParams(dto.poolId));
+    const methodAbi = this.getMethodAbi(new URLSearchParams(dto.poolId), 'TRANSFER');
     const response = await lastValueFrom(
       this.http.post<EthConnectAsyncResponse>(
-        `${this.contractInstanceUrl}/${dto.poolId}/transferWithData`,
+        `${this.baseUrl}`,
         {
-          from: dto.from,
-          to: dto.to,
-          amount: dto.amount,
-          data: encodeHex(dto.data ?? ''),
-        },
+          headers: {
+            type: sendTransactionHeader,
+          },
+          from: dto.operator,
+          to: validPoolId.address,
+          method: methodAbi,
+          params: [dto.from, dto.to, dto.amount, encodeHex(dto.data ?? '')],
+        } as EthConnectMsgRequest,
         this.postOptions(dto.operator, dto.requestId),
       ),
     );
     return { id: response.data.id };
   }
 
   async burn(dto: TokenBurn): Promise<AsyncResponse> {
+    const validPoolId: ITokenPool = this.validatePoolId(new URLSearchParams(dto.poolId));
+    const methodAbi = this.getMethodAbi(new URLSearchParams(dto.poolId), 'BURN');
     const response = await lastValueFrom(
       this.http.post<EthConnectAsyncResponse>(
-        `${this.contractInstanceUrl}/${dto.poolId}/burnWithData`,
+        `${this.baseUrl}`,
         {
-          data: encodeHex(dto.data ?? ''),
-          from: dto.from,
-          amount: dto.amount,
-        },
+          headers: {
+            type: sendTransactionHeader,
+          },
+          from: dto.operator,
+          to: validPoolId.address,
+          method: methodAbi,
+          params: [dto.from, dto.amount, encodeHex(dto.data ?? '')],
+        } as EthConnectMsgRequest,
         this.postOptions(dto.operator, dto.requestId),
       ),
     );
     return { id: response.data.id };
   }
 
   async balance(dto: TokenBalanceQuery): Promise<TokenBalance> {
+    const validPoolId: ITokenPool = this.validatePoolId(new URLSearchParams(dto.poolId));
+    const methodAbi = this.getMethodAbi(new URLSearchParams(dto.poolId), 'BALANCE');
     const response = await lastValueFrom(
-      this.http.get<EthConnectReturn>(`${this.contractInstanceUrl}/${dto.poolId}/balanceOf`, {
-        params: {
-          account: dto.account,
+      this.http.post<EthConnectAsyncResponse>(`${this.baseUrl}`, {
+        headers: {
+          type: sendTransactionHeader,
```

Comment:
Balance is intended to be a synchronous "call" to eth which returns the actual balance - not a submitted transaction. Feel free to completely omit it for now if you want - it's not a required part of the fftokens interface.

Reply:
Removed